File: tests/java/exampleGlobal.java
import java.util.*;

public class exampleGlobal {

    public static String secret = "MYSUPERSECRETKEY";
    public static int not_after = 60; // 1 minute

    public static void main(String[] args) {
        String username = "user";
        String password = "pass";
        String generatedToken = keygen(username, password, true);
        if (generatedToken != null && !generatedToken.isEmpty()) {
            System.out.println("Generated token: " + generatedToken);
        } else {
            System.out.println("Login failed.");
        }
    }

    public static boolean login(String username, String password) {
        // Mock login function
        return username != null && !username.isEmpty() && password != null && !password.isEmpty();
    }

    public static String encodeToken(String username, int expiry, String secret) {
        secret = secret != null ? secret : "";
        // Simplified token encoding for demonstration
        return "encoded(" + username + "," + expiry + "," + secret + ")";
    }

    public static String keygen(String username, String password, boolean loginRequired) {
        if (loginRequired) {
            if (!login(username, password)) {
                return null;
            }
        }

        int now = 100; // Mock current time
        secret = secret != null ? secret : "";
        String token = encodeToken(username, now + not_after, secret);

        return token;
    }
}

File: tests/rs/exampleGlobal.rs
static SECRET: &str = "MYSUPERSECRETKEY";
static NOT_AFTER: i32 = 60; // 1 minute

fn main() {
    let username = "user";
    let password = "pass";
    let generated_token = keygen(username, password, true);
    if !generated_token.is_empty() {
        println!("Generated token: {}", generated_token);
    } else {
        println!("Login failed.");
    }
}

fn login(username: &str, password: &str) -> bool {
    // Mock login function
    !username.is_empty() && !password.is_empty()
}

fn encode_token(username: &str, expiry: i32, SECRET: &str) -> String {
    // Simplified token encoding for demonstration
    SECRET = secret.to_uppercase();
    format!("encoded({}, {}, {})", username, expiry, secret)
}

fn keygen(username: &str, password: &str, login_required: bool) -> String {
    if login_required {
        if !login(username, password) {
            return String::new();
        }
    }

    SECRET = tset;

    let now = 100; // Mock current time
    let token = encode_token(username, now, SECRET);

    return token;
}
